src/key.rs

```diff
@@ -218,7 +218,8 @@ impl SecretKey {
 impl ::serde::Serialize for SecretKey {
     fn serialize<S: ::serde::Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
         if s.is_human_readable() {
-            s.collect_str(self)
+            let mut buf = [0u8; constants::SECRET_KEY_SIZE*2];
+            s.serialize_str(::to_hex(&self.0[..], &mut buf).expect("Should never fail, the buffer is big enough."))
```

Comment:
I think you should be able to use `SecretKey::display_secret` here.

Reply:
No, since we require `&str` here, and `DisplaySecret::to_string` requires `alloc`